File: python/122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans, buy = 0, -1
        for i in range(len(prices) - 1):
            if buy >=0 and  prices[i] > buy and prices[i+1] < prices[i]:
                ans += prices[i] - buy
                buy = -1
            if buy < 0 and prices[i+1] > prices[i]:
                buy = prices[i]
        
        if buy >= 0 and prices[-1] > buy:
            return ans + prices[-1] - buy
        return ans

File: hao/c++/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int last = prices[0];
        
        int maxProfit = 0;
        for (vector<int>::iterator it = prices.begin()+1; it != prices.end(); ++it) {
            if (*it > last) {
                maxProfit += *it-last;
            }
            last = *it;
        }
        return maxProfit;
    }
};